plugins/android-boot/fu-android-boot-device.c

```diff
@@ -116,10 +113,27 @@ fu_android_boot_device_probe(FuDevice *device, GError **error)
 	fu_device_add_instance_strsafe(device, "LABEL", self->label);
 	fu_device_add_instance_strsafe(device, "SLOT", self->boot_slot);
 
+	/*
+	 * eMMC boot partitions cannot have a label, UUID, etc. because they do not contain a
+	 * partition table. These devices are typically used to store platform or bootloader
+	 * firmware on it at offset 0. Combine their exposed device name, path and hardware name for
+	 * a GUID. The GUID depends on the device DT and therefore unique.
+	 */
+	fu_device_add_instance_strsafe(device,
+				       "DEVNAME",
+				       g_udev_device_get_property(udev_device, "DEVNAME"));
+	fu_device_add_instance_strsafe(device,
+				       "ID_PATH",
+				       g_udev_device_get_property(udev_device, "ID_PATH"));
```

Comment:
This isn't a great GUID as it's not part of the kernel API; the exact name/path isn't going to be predictable from the kernel. Is there anything else we can use?

Reply:
I'm afraid not... All the information to uniquely identify these partitions is not available because it has no partition table (no UUIDs, no labels, etc.). Setting these doesn't even work here as these are not 'regular' partitions. They always have `boot` in their devname if they exist on a device though.

Reply:
Could we read back the block data and read any header?

Reply:
Well we have uSWID data for Tow-Boot (already for the version information), we could extract maybe something else from that?

Reply:
That does feel a bit circular, but yes, that's probably a better idea than relying on the kernel naming scheme.

Reply:
- So extending the firmware version retrieval to find `Tow-Boot` there and use that to build GUIDs?
- I would use this as a fallback, so try to build GUIDs with the ones we have and if no GUIDs are generated then, fallback to this approach?
- For performance reasons I would do the version retrieval then also in the probe method to read the firmware only once.

Reply:
Did that fly? What's the status of this PR please?

Reply:
I haven't been able to test this out I'm afraid. Currently, I swamped with $DAYJOB. I hope to pick this up soon, but it might take several weeks.

Reply:
Got it, thanks. No panic.

Reply:
Yes, let's close this one for now. I'm still interested in this, but upstream stuff stalled (need Tow-Boot etc. on board). We can re-open later.